docs/sql-data-sources-jdbc.md

```diff
@@ -197,6 +197,20 @@ the following case-insensitive options:
      The option to enable or disable predicate push-down into the JDBC data source. The default value is true, in which case Spark will push down filters to the JDBC data source as much as possible. Otherwise, if set to false, no filter will be pushed down to the JDBC data source and thus all filters will be handled by Spark. Predicate push-down is usually turned off when the predicate filtering is performed faster by Spark than by the JDBC data source.
     </td>
   </tr>
+
+  <tr>
+    <td><code>keytab</code></td>
```

Comment:
One thing I dislike about this approach is that it forces you to change application code depending on how you want to authenticate. Have you thought about ways in which to avoid that? 

e.g. if the postgres jdbc url syntax allows for these options to be defined there, you could check whether there's a placeholder for the principal / keytab and replace with some globally-set Spark config.

Another question I have is whether postgres (at least, since that's what this change covers) can use auth data from the current logged in subject, in which case the above discussion could be solved by extending Spark's principal / keytab support to the executor processes too.

Reply:
> e.g. if the postgres jdbc url syntax allows for these options to be defined there, you could check whether there's a placeholder for the principal / keytab and replace with some globally-set Spark config.

I've considered global config but since Spark can use multiple databases at the same time like postgres on source side and oracle on sink side I thought it would make the implementation rigid.

The other option which is the jdbc url also considered but that way for each database type an url parser must be implemented which I thought is an overkill (for instance postgres uses `&` as separator and MSSQL uses `;`).

Reply:
> Another question I have is whether postgres (at least, since that's what this change covers) can use auth data from the current logged in subject, in which case the above discussion could be solved by extending Spark's principal / keytab support to the executor processes too.

I've gone through the documentation + scanned the code but not found such possibility.

Reply:
> I've gone through the documentation + scanned the client code but not found such possibility.

Sigh. Single sign-on kinda loses its meaning if every library insists on managing its own login and requiring user creds... :-/ Will have to live with it, I guess.